From: Peng Tao <tao.peng@primarydata.com>
Date: Fri, 11 Sep 2015 11:14:06 +0800
Subject: [PATCH] nfs: fix pg_test page count calculation
Git-commit: 048883e0b934d9a5103d40e209cb14b7f33d2933
Patch-mainline: v4.3
References: bnc#898675

We really want sizeof(struct page *) instead. Otherwise we limit
maximum IO size to 64 pages rather than 512 pages on a 64bit system.

Fixes 2e11f829(nfs: cap request size to fit a kmalloced page array).

Cc: Christoph Hellwig <hch@lst.de>
Signed-off-by: Peng Tao <tao.peng@primarydata.com>
Fixes: 2e11f8296d22 ("nfs: cap request size to fit a kmalloced page array")
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/nfs/pagelist.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/nfs/pagelist.c
+++ b/fs/nfs/pagelist.c
@@ -296,7 +296,7 @@ bool nfs_generic_pg_test(struct nfs_page
 	 * for it without upsetting the slab allocator.
 	 */
 	if (((desc->pg_count + req->wb_bytes) >> PAGE_SHIFT) *
-			sizeof(struct page) > PAGE_SIZE)
+			sizeof(struct page *) > PAGE_SIZE)
 		return false;
 
 	return desc->pg_count + req->wb_bytes <= desc->pg_bsize;
